### ISSUE TITLE
Colors are highlighted in JSON formatted documents

### ISSUE THREAD

**Issue by bryanbraun** [abe33/atom-pigments#93]
This is what I see.

![image](https://cloud.githubusercontent.com/assets/1256329/10152375/3d669ef6-661f-11e5-86c5-4ca9fb1309c9.png)

I only expect to see highlighted color names when the file syntax is CSS, Less, Sass, or something similar. It's possible that this is a configuration issue, but I'm using the default settings and this is a `.json` file. I went through the package settings and couldn't find any obvious way to fix it.

If it's a settings issue, then my recommendation is to improve the default settings.

**Comment by abe33** [comment]
Hi @bryanbraun, please take a look at the `Ignored Scopes` section of the README: https://github.com/abe33/atom-pigments#ignored-scopes.

**Comment by madranet** [comment]
I don't understand this either. I've read the docs and looked at the settings and [unless I'm looking at the wrong thing], it seems Pigments should only highlight colours in CSS [and similar] documents. however, I'm seeing colours highlighted in any kind of document, even plain-text.

Can you point to the specific option that controls this? 

This issue keeps coming up and you keep basically telling people to 'RTFM' and closing the issue. However the fact it keeps coming up again and again should tell you that the documentation and/or options aren't clear.

**Comment by abe33** [comment]
Sorry if the documentation isn't clear enough, but nowhere in the README I state that Pigments only works for CSS files only. It works for every files as long as you don't explicitly ignore their scope.

> Can you point to the specific option that controls this? 

Did you clicked on the link I posted above? Because it's the option that controls this!

Now if you think my explanations aren't clear enough, well, what can I say. Yeah, probably. English isn't my native language. Now it's an open source project. Go, contribute, ask if something isn't clear enoug. But don't say I'm just telling RTFM whenever someone's popup with a question already solved and answered several times. I maintain a dozen of packages on my free time, and believe it or not it's a lot of work. And even with that I try as frequently as possible to find an answer to people's problems.

Now, to come back to the problem at hand, the `Ignored Scopes` setting (linked above) takes an array of regular expressions strings. 

So if the user wants to ignore colors in a json file, he can use the `Log Cursor Scope` command to get that scope. He'll see then that json files are identified with the `.source.json` scope. To ignore these files he can just add `\.source\.json` in the setting field. That's it. 

If you want to ignore everything except css files, it's a bit more tricky, but feasible. I explained how in issue #57, but to save you the effort, you just need to use `^(?!\.source\.css).*` as expression. It matches everything except scopes that starts with `.source.css`, which means pretty much everything.

**Comment by madranet** [comment]
Thanks for taking the time to reply. ```^(?!\.source\.css).*``` did the job nicely.

I think the confusion lies with the wording in the Pigments settings window. "Ignored Scopes" doesn't exactly make a lot of sense to the 'average user', as the concept of "scopes" is probably more familiar to Atom developers [a bit like the potential confusion between 'buffers' and 'files']. Might be better to use something like ```Ignored Filetypes```. And, of course, the requirement for a bit of RegEx-fu mastery, just to set an option for "only colour CSS files" is not exactly user friendly.
>>I maintain a dozen of packages on my free time, and believe it or not it's a lot of work. And even with that I try as frequently as possible to find an answer to people's problems.

I don't doubt it for a moment. I wouldn't use Pigments, if I didn't find it useful. But, please don't fall into the trap so many developers do of equating 'raising an issue' with attacking you personally, or denigrating your work. If we didn't care about your product, we wouldn't bother suggesting improvements or raising questions about it. We do so because we want to help you make it better.

**Comment by abe33** [comment]
I started to think of a way to improve the documentation with a specific section about how to ignore files/scopes in pigments.

> "Ignored Scopes" doesn't exactly make a lot of sense to the 'average user', as the concept of "scopes" is probably more familiar to Atom developers

I'm not sure about that, the concepts are quite different, while you can use scopes to ignore a file type using its source scope, you can also use it in a more fine grained way, like the examples I've written in the README (ignore string, comments). I fear that bypassing this would lead to even more confusion. But I'll think of a way to populate this array in a more easy way.
And scopes are at the core of most Atom features (commands, keybindings, grammar, etc). There's probably some effort to produce to make that explicit from a user point of view (not only in Pigments, but in Atom as a whole).

>  But, please don't fall into the trap so many developers do of equating 'raising an issue' with attacking you personally, or denigrating your work

You're right, please forgive my rudeness at the time.

**Comment by abe33** [comment]
@madranet I'm introducing a new setting where you can use file extensions instead of regular expressions to specify the files where colors will be highlighted, let me know if the documentation in the README feels clearer for you.

**Comment by madranet** [comment]
The documentation certainly seems a lot clearer now.  

Couple of small points though:

1.  I *think* that the 'Supported Filetypes' box wasn't visible in the settings window, but appeared only after I manually added:

```
pigments:
    supportedFiletypes: [
      "css"
      "less"
    ]
```

into the config.json file. If I delete that entry from config.json, the 'Supported Filetypes' box remains visible in the settings window [albeit empty]. So it maybe that I just didn't notice it before, but I was looking specifically for it after reading your new documentation and didn't see it.

2. When I opened config.json to add the above, the previous regex code that you gave me to exclude all but css files: ```^(?!\.source\.css).*``` was listed thus in config.json:

```
ignoredScopes: [
      "^(?!\.source\.css).*"
    ]
```

When I deleted that and added it again through the settings interface, it was written to config.json as:

```
ignoredScopes: [
      "^(?!\\.source\\.css).*"
    ]
```

again, I'm pretty sure it wasn't double-backslashed before.

Hope that's of some use.

**Comment by abe33** [comment]
>  I think that the 'Supported Filetypes' box wasn't visible in the settings window, but appeared only after I manually added

Weird, are you using version 0.17? Because it should appear right below the `Source Names` setting: 

![settings_-__users_c_nehemie_development_gsoi_rep-website_-_atom](https://cloud.githubusercontent.com/assets/247140/10427081/bdfb710a-70e6-11e5-98cc-41bbcdfaced1.png)

> again, I'm pretty sure it wasn't double-backslashed before.

Right, the thing is when you define it in the settings view, it will be serialized with a double-backslash (which is the proper way to escape a backslash in a string), if you manually modify your `config.cson` file you should also use the double-backslash form, otherwise you're escaping the dot, and I'm not sure it'll behave properly when converted into a regexp. I should make that clear in the documentation though.

**Comment by madranet** [comment]
>Weird, are you using version 0.17? 

0,15. I'm not seeing any 'update available' in Atom. As I said though, I might just have missed the 'Supported Filetypes' box in the settings window, the first time I looked.

>when you define it in the settings view, it will be serialized with a double-backslash... if you manually modify your config.cson file you should also use the double-backslash form...

That's OK then. As long as the double-backslashes are the desired behaviour. I just thought that it didn't do so previously when you sent me the regex and I applied it through the settings window. But, again, I may be mistaken.

*[Now, if you just fix the docs so "colour" is spelt properly, instead of in American, it'll be 'parfait'! :-) ]*
